### ISSUE TITLE
[BUG] window_function_test hanging in databricks 10.4 runtime

### ISSUE THREAD

**Issue by pxLi** [NVIDIA/spark-rapids#6163]
**Describe the bug**
rapids_databricks_nightly-pre_release-github build 451, 452
rapids_databricks_nightly-dev-github build 170,171,172

parallel test in spark local mode

[2022-07-29T**12:41:59.014Z**] [gw3] [ 99%] PASSED ../../src/main/python/window_function_test.py::test_window_aggs_lead_ignore_nulls_fallback[partBy:Long-orderBy:Long-orderBy:LongRange(not_null)-agg:Boolean][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(WindowExec,Alias,WindowExpression,Lead,Literal,WindowSpecDefinition,SpecifiedWindowFrame)] 

[2022-07-29T**14:45:47.874Z**] [gw3] [ 99%] PASSED ../../src/main/python/window_function_test.py::test_window_aggs_lead_ignore_nulls_fallback[partBy:Long-orderBy:Long-orderBy:LongRange(not_null)-agg:Date][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(WindowExec,Alias,WindowExpression,Lead,Literal,WindowSpecDefinition,SpecifiedWindowFrame)] Sending interrupt signal to process

those pipeline failed timeout seems abnormal long test time for specific cases,
this only occurred in databricks 10.4 runtime (9.1 works fine)

**Comment by pxLi** [comment]
this also occurred intermittently in premerge CI 
rapids_premerge-github build 5273  triggered in #5989

**Comment by mythrocks** [comment]
My attempt at repro-ing this produced the same results as @tgravescs's:
- Running only `test_window_aggs_lead_ignore_nulls_fallback` didn't repro the problem, even after several runs.
- Running all (and only) the tests under `window_function_tests.py` did not reproduce the problem.

I'm now running the full set of python tests. The problem looks elusive.

**Comment by tgravescs** [comment]
this didn't seem to happen last night as our nightly passed in two hours.

**Comment by mythrocks** [comment]
Yesterday, I was able to run the integration tests to completion 3 times on Databricks 10.4. (I wish I had timed those runs, in hindsight.) The window function tests did not hang indefinitely.

I was also able to run just the window function tests to completion several times.

I wonder if what we're seeing is a performance regression somewhere that results in the full integration suite taking longer than the 2 hour timeout. (Note that the window function tests are among the very last to run.)

The hang is as yet not reproducible for me. :/
